sound/soc/sof/intel/hda-codec.c

```diff
@@ -54,8 +54,16 @@ static int request_codec_module(struct hda_codec *codec)
 
 static int hda_codec_load_module(struct hda_codec *codec)
 {
-	int ret = request_codec_module(codec);
+	int ret;
+
+	ret = snd_hdac_device_register(&codec->core);
+	if (ret) {
+		dev_err(&codec->core.dev, "failed to register hdac device\n");
+		put_device(&codec->core.dev);
+		return ret;
+	}
```

Comment:
it's hard to follow the commit message 

The current code flow is:
snd_hdac_device_register()
set parameters needed by the hdac driver <<<< I see nothing between register() and request()?
request_codec_module()


Move the snd_hdac_device_register() later, to be done right before
requesting the codec module to make sure that the parameters are all set
before the device is created. <<< again it's not clear where those parameters are set??